src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -370,6 +372,22 @@ private static void DeserializeIntoObjectProperties(this CodeWriter writer, IEnu
             }
         }
 
+        private static FormattableString GetEmptyStringCheckClause(JsonPropertySerialization property, FormattableString itemVariable, bool shouldTreatEmptyStringAsNull)
+        {
+            FormattableString result = $"";
+            if (!shouldTreatEmptyStringAsNull)
+                return result;
+            if (property.ValueSerialization is JsonValueSerialization valueSerialization
+                && valueSerialization.Type.IsFrameworkType)
+            {
+                if (Configuration.TreatEmptyStringAsNullForTypes.Contains(valueSerialization.Type.FrameworkType.Name))
+                {
+                    result = $" || {itemVariable}.Value.GetString().Length == 0";
```

Comment:
What if the `JsonValueKind` is not a string? Right now `treat-empty-string-as-null` is specified by human. If someone made a mistake (e.g. specifying a wrong property which is not of string value type), which behavior is better?
- `Value.GetString()` throw an exception
- We add extra check on the `JsonValueKind` so that we let the code continue

Reply:
The config is supposed to be used for the types that are originally string in swagger but converted to other types by the generator. If we have a way to know the original type in swagger, then we can add check for the values of this config.

Reply:
How about this check?
```suggestion
                    result = $" || (({itemVariable}.Value.ValueKind == {typeof(JsonValueKind)}.{nameof(JsonValueKind.String}} && {itemVariable}.Value.GetString().Length == 0)";
```

Reply:
Updated.